app/validators/schema_name_format_validator.rb

```diff
@@ -0,0 +1,9 @@
+class SchemaNameFormatValidator < ActiveModel::Validator
+  def validate(record)
+    if record.format.blank? && record.schema_name.blank?
+      record.errors.add(:format, 'must be supplied if schema_name is blank')
+    elsif record.schema_name.present? && record.document_type.blank?
+      record.errors.add(:document_type, 'must be supplied if schema_name is present')
+    end
```

Comment:
The third branch of this is "what if we don't have a schema name at all" and the validation for that is currently undefined.